utils/process_retry.py

```diff
@@ -0,0 +1,72 @@
+# coding=utf-8
+# Copyright 2025 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Utility functions for calling processes. Especially useful for calling processes that might fail.
+"""
+
+import argparse
+import logging
+import shlex
+import subprocess
+import time
+
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_RETRIES: int = 5
+DEFAULT_TIMEOUT: int = 1
+
+
+def call_proc(cmd) -> tuple[bytes, bytes]:
+    """Call a process and return the output and error streams."""
+    try:
+        p = subprocess.Popen(shlex.split(cmd), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out, err = p.communicate()
+        return (out, err)
+    except Exception as e:
+        return (b"", str(e).encode())
+
+
+def retry_call_proc(cmd, retries=DEFAULT_RETRIES, timeout=DEFAULT_TIMEOUT, debug=False) -> bytes | None:
```

Comment:
I think `timeout` is a bit misleading. I am not a naming expert, but I see there is `wait_before_retry` in `def is_flaky` which might be a better name